src/api/routers/gcp/chat.py

```diff
@@ -105,19 +111,70 @@ def get_chat_completion_model_name(model_alias):
 
     return model_alias.split('/')[-1]
 
+def transform_vertex_chunk_to_openai(chunk_line: str, index: int = 0) -> str:
+    if not chunk_line.startswith("data: "):
+        return ""  # skip irrelevant lines like empty keepalives
+
+    try:
+        payload = json.loads(chunk_line[len("data: "):])
+        parts = payload.get("candidates", [])[0].get("content", {}).get("parts", [])
+        if not parts:
+            return ""
+        text = parts[0].get("text", "")
+    except Exception:
+        return ""
+
+    transformed = {
+        "choices": [
+            {
+                "delta": {"content": text},
+                "index": index,
+                "finish_reason": None
+            }
+        ]
+    }
+
+    return f"data: {json.dumps(transformed)}\n"
+
+async def stream_generator(target_url: str, request_headers: dict, content_json: dict, model_alias: str) -> AsyncGenerator[str, None]:
+    async with httpx.AsyncClient(timeout=None) as client:
+        async with client.stream(
+            "POST",
+            target_url,
+            headers=request_headers,
+            json=content_json,
+        ) as response:
+
+            async for line in response.aiter_lines():
+                if not line.strip():
+                    continue
+
+                if line.strip() == "data: [DONE]":
+                    yield openai_done()
+                    break
+
+                if line.startswith("data: "):
+                    raw_json = line[6:].strip()
```

Comment:
I don't like this magic number. I'm also tempted to factor out a constant string `EVENT_PREFIX = "data: "` and re-use it here and in `transform_vertex_chunk_to_openai`

```suggestion
                    raw_json = line[len("data: "):].strip()
```